src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -252,27 +294,144 @@ void Engine::ApplyPerformanceHints(std::map<std::string, std::string> &config, c
-        config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] = std::to_string(tput_hints.second.threads_per_stream_big);
+    if (cpu_map_available()) {
+        int streams = engConfig.streamExecutorConfig._streams;
+        if (perf_hint_name == CONFIG_VALUE(LATENCY)) {
+            streams = static_cast<int>(getAvailableNUMANodes().size());
+        } else if (perf_hint_name == CONFIG_VALUE(THROUGHPUT)) {
+            streams = 0;
+        } else if (perf_hint_name.empty()) {
+            streams = streams > 1 ? streams : (engConfig.streamExecutorConfig._set_streams ? streams : 0);
+        }
+        const auto common_hints = getNumStreams(streams);
+
+        config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = common_hints.first;
+        config[ov::num_streams.name()] = common_hints.first;
+        config[CONFIG_KEY(CPU_THREADS_NUM)] = std::to_string(common_hints.second.num_threads);
+        config[CONFIG_KEY_INTERNAL(BIG_CORE_STREAMS)] = std::to_string(common_hints.second.big_core_streams);
+        config[CONFIG_KEY_INTERNAL(BIG_CORE_LOGIC_STREAMS)] = std::to_string(common_hints.second.big_core_logic_streams);
+        config[CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)] = std::to_string(common_hints.second.small_core_streams);
+        config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] =
+            std::to_string(common_hints.second.threads_per_stream_big);
         config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)] =
-            std::to_string(tput_hints.second.threads_per_stream_small);
-        config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(tput_hints.second.small_core_offset);
+            std::to_string(common_hints.second.threads_per_stream_small);
+        config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(common_hints.second.small_core_offset);
+    } else {
+        if (perf_hint_name == CONFIG_VALUE(LATENCY)) {
+            config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = latency_hints.first;
+            config[ov::num_streams.name()] = latency_hints.second;
+        } else if (perf_hint_name == CONFIG_VALUE(THROUGHPUT)) {
+            config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = tput_hints.first;
+            config[ov::num_streams.name()] = tput_hints.first;
+            config[CONFIG_KEY_INTERNAL(BIG_CORE_STREAMS)] = std::to_string(tput_hints.second.big_core_streams);
+            config[CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)] = std::to_string(tput_hints.second.small_core_streams);
+            config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] =
+                std::to_string(tput_hints.second.threads_per_stream_big);
+            config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)] =
+                std::to_string(tput_hints.second.threads_per_stream_small);
+            config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(tput_hints.second.small_core_offset);
+        }
     }
 }
 
+int Engine::GetModelPreferThreads(const int num_streams,
```

Comment:
It looks like copypaste of getNumStreamsThroughput. Why do we need both?

Reply:
Don't need both. I created a function `GetPerformanceStreams` separately. And moved calculation streams logic to `cpu_streams_caclutalition`.